### PR TITLE
tools: fix store update flow

### LINKED ISSUE #5796: tools: hot read is empty when pd-heartbeat-bench
### What did you do?

pd-heartbeat-bench 

### What did you expect to see?
report hot read flow
### What did you see instead?
image

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Close #5796

### What is changed and how does it work?

update store info

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
image

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-heartbeat-bench/main.go [hunk 1]
```diff
@@ -333,11 +330,32 @@ func (rs *Regions) handleRegionHeartbeat(wg *sync.WaitGroup, stream pdpb.PD_Regi
...
+func (s *Stores) heartbeat(ctx context.Context, cli pdpb.PDClient, storeID uint64) {
+	s.RLock()
+	defer s.RUnlock()
+	cctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	cli.StoreHeartbeat(cctx, &pdpb.StoreHeartbeatRequest{Header: header(), Stats: s.stat[storeID]})
+}
+
+func (s *Stores) update(rs *Regions) {
+	s.Lock()
```
Comment (nolouch) [inline tools/pd-heartbeat-bench/main.go:354]: Will this lock be bottleneck in large number regions benchmark?

**Comment by lhy1024** [comment]
Test Result is similiar

Config
```
round = 5

store-count = 100
region-count = 2000000

key-length = 56
replica = 3

leader-update-ratio = 0.06
epoch-update-ratio = 0.04
space-update-ratio = 0.15
flow-update-ratio = 0.35

sample = false

```

This PR
```
[2022/12/23 11:05:53.017 +08:00] [INFO] [main.go:330] ["store finish one round region heartbeat"] [store-id=42] [cost-time=27.247004525s]
[2022/12/23 11:05:53.028 +08:00] [INFO] [main.go:330] ["store finish one round region heartbeat"] [store-id=17] [cost-time=27.238652607s]
[2022/12/23 11:05:53.301 +08:00] [INFO] [main.go:526] ["update speed of each category"] [rps=70470.9481] [save-tree=4228.2569] [save-kv=2818.8379] [save-space=10570.6422] [save-flow=24664.8318] [skip=35127.3011]
[2022/12/23 11:05:53.489 +08:00] [INFO] [main.go:471] ["region heartbeat stats"] [total=28.3165s] [slowest=27.2622s] [fastest=0.0000s] [average=12.3902s] [stddev=7.7833s] [rps=70630.0907]
[2022/12/23 11:05:53.489 +08:00] [INFO] [main.go:478] ["store heartbeat stats"] [max=28.3805s]

```

Master
```
[2022/12/23 11:15:28.213 +08:00] [INFO] [main.go:333] ["store finish one round region heartbeat"] [store-id=53] [cost-time=26.824179364s]
[2022/12/23 11:15:28.220 +08:00] [INFO] [main.go:333] ["store finish one round region heartbeat"] [store-id=12] [cost-time=26.831635473s]
[2022/12/23 11:15:28.483 +08:00] [INFO] [main.go:506] ["update speed of each category"] [rps=69294.9857] [save-tree=4157.6991] [save-kv=2771.7994] [save-space=10394.2479] [save-flow=24253.2450] [skip=34541.1251]
[2022/12/23 11:15:28.531 +08:00] [INFO] [main.go:452] ["region heartbeat stats"] [total=28.8621s] [slowest=26.8316s] [fastest=0.0000s] [average=12.5618s] [stddev=7.6409s] [rps=69294.9594]
[2022/12/23 11:15:28.531 +08:00] [INFO] [main.go:459] ["store heartbeat stats"] [max=28.8621s]
```

File: tools/pd-heartbeat-bench/main.go (hunk 1 above)
Comment (lhy1024) [reply tools/pd-heartbeat-bench/main.go:354]: Their result is similar with 2000000 regions

**Comment by lhy1024** [comment]
## Test with atomic
### 5000000 Region

Config
```
round = 5

store-count = 100
region-count = 5000000

key-length = 56
replica = 3

leader-update-ratio = 0.06
epoch-update-ratio = 0.04
space-update-ratio = 0.15
flow-update-ratio = 0.35

sample = false

```

This PR
```
[2022/12/23 18:40:59.558 +08:00] [INFO] [main.go:331] ["store finish one round region heartbeat"] [store-id=48] [cost-time=2m37.603795054s]
[2022/12/23 18:40:59.565 +08:00] [INFO] [main.go:331] ["store finish one round region heartbeat"] [store-id=43] [cost-time=2m37.611398336s]
[2022/12/23 18:41:00.370 +08:00] [INFO] [main.go:471] ["region heartbeat stats"] [total=162.9503s] [slowest=157.6114s] [fastest=0.0000s] [average=75.5986s] [stddev=47.1155s] [rps=30684.2112]
[2022/12/23 18:41:00.370 +08:00] [INFO] [main.go:478] ["store heartbeat stats"] [max=162.9503s]
```

Master
```
 [2022/12/23 18:47:10.823 +08:00] [INFO] [main.go:333] ["store finish one round region heartbeat"] [store-id=11] [cost-time=2m39.993081695s]
[2022/12/23 18:47:10.873 +08:00] [INFO] [main.go:333] ["store finish one round region heartbeat"] [store-id=93] [cost-time=2m39.944812758s]
[2022/12/23 18:47:11.917 +08:00] [INFO] [main.go:452] ["region heartbeat stats"] [total=162.3355s] [slowest=159.9931s] [fastest=0.0000s] [average=75.2664s] [stddev=46.7769s] [rps=30800.4084]
[2022/12/23 18:47:11.917 +08:00] [INFO] [main.go:459] ["store heartbeat stats"] [max=162.3355s]

```

**Review by nolouch — APPROVED** [review]
lgtm

**Review by rleungx — APPROVED** [review]

**Pull request merged**